code/datums/components/growth_and_differentiation.dm

```diff
@@ -44,6 +41,8 @@
 	src.optional_checks = optional_checks
 	src.optional_grow_behavior = optional_grow_behavior
 
+	RegisterSignal(parent, COMSIG_COMPONENT_KILL, PROC_REF(kill_processes))
```

Comment:
if you don't want to use get component, allow users of the component to pass in signals that trigger the component disabling/deleting itself